elftools/elf/elffile.py

```diff
@@ -106,12 +106,23 @@ def get_section_by_name(self, name):
         # mapping
         #
         if self._section_name_map is None:
-            self._section_name_map = {}
-            for i, sec in enumerate(self.iter_sections()):
-                self._section_name_map[sec.name] = i
+            self._make_section_name_map()
         secnum = self._section_name_map.get(name, None)
         return None if secnum is None else self.get_section(secnum)
 
+    def get_section_index(self, section_name):
+        """ Gets the index of the section by name. Return None if no such
+            section name exists.
+        """
+        # The first time this method is called, construct a name to number
+        # mapping
+        #
+        if self._section_name_map is None:
+            self._make_section_name_map()
+        if not self._section_name_map.has_key(section_name):
+            return None
+        return self._section_name_map[section_name]
```

Comment:
I think you could put this into a single statement (like in the function above, line 110):

```python
return self._section_name_map.get(section_name, None)
```